### PR TITLE
Moves ModelParamTraversal to traverse_util and updates references

### PR DESCRIPTION
**Pull request by marcvanzee**

### CODE REVIEW

File: tests/optim_test.py [hunk 1]
```diff
@@ -94,7 +94,7 @@ def test_empty_optimizer(self):
 class ModelParamTraversalTest(absltest.TestCase):
 
   def test_only_works_on_models(self):
-    traversal = optim.ModelParamTraversal(lambda *_: True)
+    traversal = traverse_util.ModelParamTraversal(lambda *_: True)
```
Comment (jheek) [inline tests/optim_test.py:97]: should this test go to traverse_util_test?

**Comment by avital** [comment]
> Btw is this a breaking change? Or should ModelParamTraversal still be exposed as optim.ModelParamTraversal
I propose we re-export it as `optim.ModelParamTraversal` but in a way that is marked as deprecated and prints a warning

**Comment by avital** [comment]
@marcvanzee Looks like this PR has been around for a while -- would you like to get it merged, or alternatively we can close the issue if you're not working on it.

**Comment by marcvanzee** [comment]
Closing this PR since I'm not working on it and the code is quite outdated.

**Pull request closed without merging**